gradle/lib/dependabot/gradle/file_updater/wrapper_updater.rb

```diff
@@ -0,0 +1,161 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "shellwords"
+
+require "dependabot/gradle/distributions"
+
+module Dependabot
+  module Gradle
+    class FileUpdater
+      class WrapperUpdater
+        extend T::Sig
+        include Dependabot::Gradle::Distributions
+
+        sig { params(dependency_files: T::Array[Dependabot::DependencyFile], dependency: Dependabot::Dependency).void }
+        def initialize(dependency_files:, dependency:)
+          @dependency_files = dependency_files
+          @dependency = dependency
+          @target_files = T.let(
+            %w(
+              /gradlew
+              /gradlew.bat
+              /gradle/wrapper/gradle-wrapper.properties
+              /gradle/wrapper/gradle-wrapper.jar
+            ),
+            T::Array[String]
+          )
+          @build_files = T.let(
+            %w(
+              build.gradle
+              build.gradle.kts
+              settings.gradle
+              settings.gradle.kts
+            ),
+            T::Array[String]
+          )
+        end
+
+        sig { params(build_file: Dependabot::DependencyFile).returns(T::Array[Dependabot::DependencyFile]) }
+        def update_files(build_file)
+          local_files = dependency_files.select do |file|
+            file.directory == build_file.directory && target_file?(file)
+          end
+
+          # If we don't have any files in the build files don't generate one
+          return dependency_files unless local_files.any?
+
+          updated_files = dependency_files.dup
+          SharedHelpers.in_a_temporary_directory do |temp_dir|
+            populate_temp_directory(temp_dir)
+            cwd = File.join(temp_dir, base_path(build_file))
+
+            # Create gradle.properties file with proxy settings
+            # Would prefer to use command line arguments, but they don't work.
```

Comment:
What command line arguments don't work?

Reply:
This is actually coming from the original `lockfile_updater`:
https://github.com/dependabot/dependabot-core/blob/main/gradle/lib/dependabot/gradle/file_updater/lockfile_updater.rb#L37

Since we agreed duplicating the code, I also copied this.

I did researched a bit on this, and it seems that if you pass the `systemProp.http.proxyHost=` related params as environment variables, it's not working for the `gradlew` download logic (the logic that downloads the gradle distribution binary). 
So the workaround was to dump it in a `gradle.properties` file, that seems to be supported